ts/utils/__tests__/number.test.ts

```diff
@@ -29,3 +29,21 @@ describe("roundToThirdDecimal", () => {
     expect(roundToThirdDecimal(parseInt("hello", 10))).toEqual(0);
   });
 });
+
+describe("bytesToGigabytes", () => {
+  it("should convert bytes to gigabytes and round to 1 decimal", () => {
+    expect(bytesToGigabytes(1000000000)).toEqual(1.0);
+  });
+
+  it("should convert bytes to gigabytes and round to 1 decimal", () => {
+    expect(bytesToGigabytes(5368709120)).toEqual(5.4);
+  });
+
+  it("should return 0 undefined is provided", () => {
+    expect(bytesToGigabytes(NaN)).toEqual(0);
+  });
+
+  it("should return 0 if not a number is computed", () => {
+    expect(bytesToGigabytes(parseInt("hello", 10))).toEqual(0);
+  });
```

Comment:
What about negative numbers, 0 and and decimal numbers between 0 and 1 (excluded)? Should we have a test also for this cases? What do you think?

Reply:
This comment made me realise that this function wasn't considering a few edge cases. I rewrote it completely and also added the possibility to render a string with the most appropriate unit, from bytes to terabytes. Yes, we still round the float value but I think that's a good compromise anyway instead of pulling out two substrings and taking only the first digit of the decimal part. I also added the suggested tests. Let me know what you think.

Reply:
Superb!